ra/registration-authority.go

```diff
@@ -77,19 +77,27 @@ func NewRegistrationAuthorityImpl(clk clock.Clock, logger *blog.AuditLogger, sta
 	return ra
 }
 
+var errUnparseableEmail = errors.New("not a valid e-mail address")
+var errEmptyDNSResponse = errors.New("empty DNS response")
+var errDNSTimeout = errors.New("DNS timeout")
+var errDNSError = errors.New("DNS error")
+
 func validateEmail(address string, resolver core.DNSResolver) (rtt time.Duration, err error) {
 	_, err = mail.ParseAddress(address)
 	if err != nil {
-		err = core.MalformedRequestError(fmt.Sprintf("%s is not a valid e-mail address", address))
-		return
+		return time.Duration(0), errUnparseableEmail
 	}
 	splitEmail := strings.SplitN(address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
-	var mx []string
-	mx, rtt, err = resolver.LookupMX(domain)
-	if err != nil || len(mx) == 0 {
-		err = core.MalformedRequestError(fmt.Sprintf("No MX record for domain %s", domain))
-		return
+	result, rtt, err := resolver.LookupHost(domain)
+	if err == nil && len(result) == 0 {
```

Comment:
We have a similar method in the VA, `problemDetailsFromDNSError`. It might be nice to consolidate these into a single function under `core/dns.go` since this adds nice test-ability.